portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,46 +29,64 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private static final String CONTENT_PARAMETER = "content";
-  private static final String TIMESTAMP_PARAMETER = "timestamp";
-  private static final String COMMENT_PARAMETER = "Comment";
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort(TIMESTAMP_PARAMETER, SortDirection.DESCENDING);
-
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery res = datastore.prepare(query);
-
-    List<Comment> comments = new ArrayList<>();
-    for(Entity entity : res.asIterable()) {
-        String content = (String) entity.getProperty(CONTENT_PARAMETER);
-        long timestamp = (long) entity.getProperty(TIMESTAMP_PARAMETER);
+    private static final String content = "content";
+    private static final String timestamp = "timestamp";
+    private static final String COMMENT_PARAMETER = "Comment";
+    private static final int defaultMaxComments = 10;
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      Gson gson = new Gson();
+      int limit = getRequestNum(request);
+      ArrayList<Comment> comments = getComments(limit);
 
-        Comment comment = new Comment(content, timestamp);
-        comments.add(comment);
+      response.setContentType("application/json;");
+      response.getWriter().println(gson.toJson(comments));
     }
-    Gson gson = new Gson();
 
-    response.setContentType("application/json;");
-    response.getWriter().println(gson.toJson(comments));
-  }
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String text = request.getParameter("comment-input");
+        long time = System.currentTimeMillis();
 
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String content = request.getParameter("comment-input");
-    long timestamp = System.currentTimeMillis();
+        Entity comment = new Entity("Comment");
+        comment.setProperty(content, text);
+        comment.setProperty(timestamp,time);
 
-    Entity comment = new Entity(COMMENT_PARAMETER);
-    comment.setProperty(CONTENT_PARAMETER, content);
-    comment.setProperty(TIMESTAMP_PARAMETER, timestamp);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(comment);
+        int max = getRequestNum(request);  
+        response.sendRedirect("/index.html?limit=" + max);
```

Comment:
So my understanding is that the limit parameter is only read by GET and POST requests to the /data endpoint, and that /index.html does not know anything about a limit parameter. So I don't think this does anything?